CAFAna/Core/SpectrumLoader.cxx

```diff
@@ -279,34 +285,93 @@ void SpectrumLoader::HandleFile(TFile *f, Progress *prog) {
                      &XSSyst_cv_tmp[syst_it]);
   }
 
+  // If we are in the FD then these variables are nonsense.
+  if (sr.isFD) {
+    sr.muon_contained = -1;
+    sr.muon_tracker = -1;
+  }
+
   int Nentries = tr->GetEntries();
-  if (max_entries != 0 && max_entries < Nentries)
+  if (max_entries != 0 && max_entries < Nentries) {
     Nentries = max_entries;
+  }
+
+  TTree *potFriend = 0;
+  f->GetObject("OffAxisWeightFriend", potFriend);
+  if (potFriend) {
+    tr->AddFriend(potFriend);
+    SetBranchChecked(potFriend, "perPOT", &sr.perPOTWeight);
+    SetBranchChecked(potFriend, "perFile", &sr.perFileWeight);
+    SetBranchChecked(potFriend, "massCorr", &sr.NDMassCorrWeight);
+    SetBranchChecked(potFriend, "specRunWght", &sr.SpecialRunWeight);
+    SetBranchChecked(potFriend, "specRunId", &sr.SpecialHCRunId);
+ 
+    std::cout << "[INFO]: Found Off axis weight friend tree "
+                 "in input file, hooking up!"
+              << std::endl;
+  } else {
+    std::cout << "[WARNING]: Off axis weightings NOT being set." << std::endl;
+    sr.perPOTWeight = 1;
+    sr.perFileWeight = 1;
+    sr.NDMassCorrWeight = 1;
+    sr.SpecialRunWeight = 1;
+    sr.SpecialHCRunId = 293;
+  }
```

Comment:
Isn't this the stuff that was supposed to be getting removed?

Reply:
So I think this is the last remnants of the PRISM stuff. The bit that was removed was the ugly opening of an external root file. This is just reading variables from a TTree that exists for the ND CAF files we use. Ideally we will remove this so we don't have to go through this annoying pre-processing stage for the ND CAFs where the extra TTree is added, but it might take a bit of time to implement. It is only relevant if the TTree is actually there, so it does not affect anything else. I should remove the cout for when the TTree is not there. But do you want this all removed before the merge?

Reply:
I missed that this had an `if()` so it doesn't affect anyone without that tree. It will still break in the srproxy world. So I think you can keep it here, but I will remove it again when I merge srproxy, so you will be stuck on the (new) prism branch until that is sorted out.